core/src/main/java/org/fao/geonet/api/records/attachments/CMISStore.java

```diff
@@ -586,7 +586,7 @@ private MetadataResourceExternalManagementProperties getMetadataResourceExternal
 
             if (metadataResourceExternalManagementPropertiesUrl.contains("{lang}") || metadataResourceExternalManagementPropertiesUrl.contains("{ISO3lang}")) {
                 final IsoLanguagesMapper mapper = ApplicationContextHolder.get().getBean(IsoLanguagesMapper.class);
-                String contextLang = context.getLanguage() == null ? Geonet.DEFAULT_LANGUAGE : context.getLanguage();
+                String contextLang = (context.getLanguage() == null || "?".equals(context.getLanguage())) ? Geonet.DEFAULT_LANGUAGE : context.getLanguage();
```

Comment:
If we are going to use "?" then I would recommend using a global. Otherwise it does not seem to be clear as to why we are checking for "?"? .  i.e. I would think the following would be clearer `Geonet.INIT_LANGUAGE.equals(context.getLanguage()`

Reply:
I am fine with that, `?` was used before I got here - having a global is fine (but we should be careful not to assume `==` )